### PR TITLE
GEN-2161: new design - Buttons

### PR DESCRIPTION
**Pull request by sladan-hedvig**
## [APP-XXX]

- Updated buttons

## Checklist

- [ ] Dark/light mode verification
- [ ] Landscape verification
- [ ] iPad verification
- [ ] iPod/iPhone 12 mini/iPhone SE verification

### CODE REVIEW

File: Projects/hCoreUI/Sources/hButton/hButton.swift [hunk 1]
```diff
@@ -294,72 +204,73 @@ struct LoaderOrContent<Content: View>: View {
...
     }
 }
 
-private struct EnvironmentHButtonIsLoading: EnvironmentKey {
-    static let defaultValue = false
-}
+struct ButtonFilledStyle: SwiftUI.ButtonStyle {
+    fileprivate var size: ButtonSize
+    @Environment(\.hButtonConfigurationType) var hButtonConfigurationType
+    @Environment(\.hUseNewDesign) var hUseNewDesign
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/hButton/hButton.swift:224]: can we remove this?

File: Projects/hCoreUI/Sources/hButton/hButton.swift [hunk 2]
```diff
@@ -294,72 +204,73 @@ struct LoaderOrContent<Content: View>: View {
...
-        self.environment(\.hButtonDontShowLoadingWhenDisabled, show)
-    }
-}
+    @ViewBuilder
+    private func getView(configuration: Configuration) -> some View {
+        if hUseNewDesign {
+            VStack {
+                Label(configuration: configuration)
+            }
+            .buttonSizeModifier(size, useNewDesing: hUseNewDesign)
```
Comment (julia-andersson) [inline Projects/hCoreUI/Sources/hButton/hButton.swift:257]: remove new design

**Review by julia-andersson — CHANGES_REQUESTED** [review]

**Comment by julia-andersson** [comment]
You could also just merge radio option branch to get rid of that hUseNewDesign

File: Projects/hCoreUI/Sources/hButton/hButton.swift (hunk 2 above)
Comment (sladan-hedvig) [reply Projects/hCoreUI/Sources/hButton/hButton.swift]: Removed!

File: Projects/hCoreUI/Sources/hButton/hButton.swift (hunk 1 above)
Comment (sladan-hedvig) [reply Projects/hCoreUI/Sources/hButton/hButton.swift]: removed new design!

**Review by julia-andersson — APPROVED** [review]

**Pull request merged**